CMakeLists.txt

```diff
@@ -221,6 +223,24 @@ install(
 
 find_package(CycloneDDS REQUIRED)
 
+get_target_property(cyclone_support_shm
+    CycloneDDS::ddsc SHM_SUPPORT_IS_AVAILABLE)
+if(cyclone_support_shm)
+  set(ENABLE_SHM ON)
+  set(DDSCXX_HAS_SHM "1")
+  message(STATUS "CycloneDDSCXX is build with shared memory support!")
+elseif(ENABLE_SHM AND NOT cyclone_support_shm)
+  message(FATAL_ERROR
+      "SHM support in CycloneDDSCXX is requested, but CycloneDDS is not build with shared memory support!")
+endif()
+
+if (ENABLE_SHM)
+  find_package(iceoryx_binding_c REQUIRED)
+endif()
```

Comment:
I was thinking of adding this to *cyclonedds*. Makes a bit more sense right?

```
set(ENABLE_SHM "auto" CACHE STRING "Enable shared memory support")
if(ENABLE_SHM)
  mark_as_advanced(find_iceoryx_flags)
  if(ENABLE_SHM MATCHES "[aA][uU][tT][oO]")
    set(find_iceoryx_flags QUIET)
  else()
    set(find_iceoryx_flags REQUIRED)
  endif()
  find_package(iceoryx_binding_c ${find_iceoryx_flags})
  if(iceoryx_binding_c_FOUND)
    set(ENABLE_SHM ON)
    set(DDS_HAS_SHM "1")
  else()
    set(ENABLE_SHM OFF)
  endif()
endif()
```

In this case you can incorporate the `SHM_SUPPORT_IS_AVAILABLE` into the logic like:
* If `ENABLE_SHM` is `OFF`, don't do any checks.
* If `ENABLE_SHM` is `ON`, require both shm in Cyclone DDS and require Iceoryx to be found.
* If `ENABLE_SHM` is `auto`, and both are true, set it to `ON`, otherwise `OFF`.

Reply:
Absolutely, makes more sense. I will fix this (also thanks for showing me how to do this correctly :))